homeassistant/components/foscam/__init__.py

```diff
@@ -44,20 +43,12 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     return True
 
 
-async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+async def async_unload_entry(hass: HomeAssistant, entry: FoscamConfigEntry) -> bool:
     """Unload a config entry."""
-    unload_ok = await hass.config_entries.async_unload_platforms(entry, PLATFORMS)
-    if unload_ok:
-        hass.data[DOMAIN].pop(entry.entry_id)
+    return await hass.config_entries.async_unload_platforms(entry, PLATFORMS)
 
-        if not hass.data[DOMAIN]:
-            hass.services.async_remove(domain=DOMAIN, service=SERVICE_PTZ)
-            hass.services.async_remove(domain=DOMAIN, service=SERVICE_PTZ_PRESET)
```

Comment:
Note: we no longer recommend removing services on unload, so I simply removed this code and didn't attempt to move it elsewhere